### ISSUE TITLE
Nested modular fields

### ISSUE THREAD

**Issue by ovsw** [datocms/product-roadmap#142]
**Is your feature request related to a problem? Please describe.**
I'd need to create more complex blocks than just the basic types offered. For example an Accordion block. I thought I just needed to create a new block in the content modular field's schema and set it to type 'modular'  but it's not available. You can only add modular fields in the root of a model.

**Describe the solution you'd like**
I'd like to be able to add a field of type "modular content" in another modular content field's block.

**Describe alternatives you've considered**
The only workaround right now would be to use a JSON field instead, which is horrible for editors to work with.

Thanks!

**Comment by manchoz** [comment]
+1

**Comment by sbc3** [comment]
This would be awesome. In our case we can get around it for the time being by using linked records but it's not ideal having to create global models for things that are only used in modular content.

**Comment by anthonysapp** [comment]
In my case I'd be more interested in a "repeater" within a modular content field for cases like the above - image galleries, content columns etc.

**Comment by ovsw** [comment]
@anthonysapp isn't a repeater just a modular field but with only one type of block?

**Comment by anthonysapp** [comment]
Yes. I guess my meaning was if there was hesitation to add the modular block within a modular block, I'd be okay with a simpler repeater.

**Comment by lunelson** [comment]
I very much agree re: repeater fields, my ultimate version of this would be to see it as an option on all the basic field types, where checking "repeatable" would make it in to an array of that type, in the response

**Comment by hellojebus** [comment]
I upvote repeater fields!

**Comment by matjack1** [comment]
Just a small note on the "repeater field". While I agree that would be handier to have the repeater option in the field you can use the modular block with one single field inside and it won't count against your number of models. So while it's slightly more complex there isn't any additional cost in doing that. But we see the point of that.

**Comment by sbc3** [comment]
> Just a small note on the "repeater field". While I agree that would be handier to have the repeater option in the field you can use the modular block with one single field inside and it won't count against your number of models. So while it's slightly more complex there isn't any additional cost in doing that. But we see the point of that.

@matjack1 that's fine for top level models, but you can't use modular content within other modular content types :(

**Comment by matjack1** [comment]
true @sbc3 that only applies on "repeater fields" in general, the nested modular content problem remains

**Comment by ItsLeeOwen** [comment]
The feature request for nested modular content is dated Feb, is this accepted as a roadmap feature, is there a milestone date for implementation?

**Comment by matjack1** [comment]
@ItsLeeOwen we agree that this feature is useful, but it's not yet in the roadmap, sorry: https://github.com/datocms/product-roadmap/projects/2

**Comment by digitalwert-developer** [comment]
Nested Modular Fields would be quite important for realizing Layouts for our Customers and their Editrs. We simply can't sell them pure Json Fields for that matter.

**Comment by digitalwert-developer** [comment]
Dear @matjack1 please put the nested feature on the roadmap. It is a key feature in many traditionell CMS. You can take any Wordpress Page builder for example (it is possible to nest sections, rows, content block). For us it gets quite hard to sell your really great CMS to a customer without this feature. 

If a customer wants to build complex layouts for a webpage, this feature is necessary, otherwise the developer hast to build an enormous amount of content blocs. It is not really an option to use models for more flexibility, because it gets confusing very fast. (Where can i edit the content for this page, just go to this module and to this module and then to this module, we got very bad customer feedback for this solution in the past)

I don't want to say the the following Headless CMS Systems are better, but they have this feature implemented quite well. Prismic or Flamelink

So it would be great to hear from you. Thanks and keep on the good work.

**Comment by matjack1** [comment]
thank you @digitalwert-developer we know this is a quite important feature, but it involves many changes at the moment. We'll surely consider this for the next iteration of development, but until November-ish we'll stick to our current roadmap: https://www.datocms.com/blog/what-happened-in-last-three-months#future-plans

We'll really consider this in the near future but we already have other features that are highly requested in the pipeline: https://github.com/datocms/product-roadmap/projects/2

We know how much this is important for editors though, it's just that we need to find a balance in introducing the right features both for you and for our internal development. This is high on the list anyway, hopefully we'll be in touch soon!

**Comment by matjack1** [comment]
This feature request has been moved to https://community.datocms.com/t/283
